fcrepo-kernel-impl/src/test/java/org/fcrepo/kernel/impl/models/ResourceFactoryImplTest.java

```diff
@@ -418,6 +419,67 @@ public void doesResourceExist_Exception_WithoutSession() throws Exception {
         factory.doesResourceExist(null, fedoraMementoId);
     }
 
+    @Test
+    public void getChildren_NoChildren() throws Exception {
+        populateHeaders(resourceHeaders, BASIC_CONTAINER);
+
+        final var childrenStream = factory.getChildren(sessionId, fedoraId);
+
+        assertEquals(0, childrenStream.count());
+    }
+
+    @Test
+    public void getChildren_DoesNotExist() throws Exception {
+        final var childrenStream = factory.getChildren(sessionId, fedoraId);
+        assertEquals(0, childrenStream.count());
+    }
+
+    @Test
+    public void getChildren_WithChildren() throws Exception {
+        populateHeaders(resourceHeaders, BASIC_CONTAINER);
+
+        final var child1IdStr = FEDORA_ID_PREFIX + "/" + UUID.randomUUID().toString();
+        final var child1Id = FedoraId.create(child1IdStr);
+        final var child1Headers = new ResourceHeadersImpl();
+        child1Headers.setId(child1Id);
+        populateHeaders(child1Headers, BASIC_CONTAINER);
+        when(psSession.getHeaders(child1Id, null)).thenReturn(child1Headers);
+
+        final var childNestedIdStr = FEDORA_ID_PREFIX + "/" + UUID.randomUUID().toString();
+        final var childNestedId = FedoraId.create(childNestedIdStr);
+        final var childNestedHeaders = new ResourceHeadersImpl();
+        childNestedHeaders.setId(childNestedId);
+        populateHeaders(childNestedHeaders, BASIC_CONTAINER);
+        when(psSession.getHeaders(childNestedId, null)).thenReturn(childNestedHeaders);
+
+        final var child2IdStr = FEDORA_ID_PREFIX + "/" + UUID.randomUUID().toString();
+        final var child2Id = FedoraId.create(child2IdStr);
```

Comment:
Same as above, don't need to prefix with `FEDORA_ID_PREFIX + "/"`, can reduce to just FedoraId.create() with the UUID.